File: 0158A.Next.Round/solution.py
def solution():
    participant_count, candidate_count = [int(x) for x in input().split()]
    scores = [int(x) for x in input().split()]

    participant_index = candidate_count - 1
    threshold = scores[participant_index]
    if scores[participant_index] > 0:
        while participant_index < participant_count and scores[participant_index] == threshold:
            participant_index += 1
    else:
        while participant_index >= 0 and scores[participant_index] == 0:
            participant_index -= 1
        participant_index += 1

    print(participant_index)


if __name__ == "__main__":
    solution()

File: 0158A.Next.Round/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var reader *bufio.Reader = bufio.NewReader(os.Stdin)
var writer *bufio.Writer = bufio.NewWriter(os.Stdout)

func Printf(format string, x ...interface{}) {
	fmt.Fprintf(writer, format, x...)
}

func Scanf(format string, x ...interface{}) {
	fmt.Fscanf(reader, format, x...)
}

func Solution() {
	var participantCount, candidateCount int
	Scanf("%d %d\n", &participantCount, &candidateCount)

	scores := make([]int, participantCount)
	for partitipantIndex := 0; partitipantIndex < participantCount; partitipantIndex++ {
		Scanf("%d", &scores[partitipantIndex])
	}

	participantIndex := candidateCount - 1
	threshold := scores[participantIndex]
	if scores[participantIndex] > 0 {
		for participantIndex < participantCount && scores[participantIndex] == threshold {
			participantIndex++
		}
	} else {
		for participantIndex >= 0 && scores[participantIndex] == 0 {
			participantIndex--
		}
		participantIndex++
	}

	Printf("%d\n", participantIndex)
}

func main() {
	Solution()
	writer.Flush()
}
